pkgs/development/haskell-modules/with-packages-wrapper.nix

```diff
@@ -25,15 +29,20 @@ assert stdenv.lib.versionOlder "6.12" ghc.version;
 #   fi
 
 let
-  ghc761OrLater = stdenv.lib.versionOlder "7.6.1" ghc.version;
+  ghc761OrLater = versionOlder "7.6.1" ghc.version;
   packageDBFlag = if ghc761OrLater then "--global-package-db" else "--global-conf";
   libDir        = "$out/lib/ghc-${ghc.version}";
   docDir        = "$out/share/doc/ghc/html";
   packageCfgDir = "${libDir}/package.conf.d";
-  paths         = stdenv.lib.filter (x: x ? isHaskellLibrary) (stdenv.lib.closePropagation packages);
-  hasLibraries  = stdenv.lib.any (x: x.isHaskellLibrary) paths;
+  paths         = filter (x: x ? isHaskellLibrary) (closePropagation packages);
+  hasLibraries  = any (x: x.isHaskellLibrary) paths;
+  # CLang is needed on Darwin for -fllvm to work:
+  # https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/code-generators.html
+  llvm          = makeSearchPath "bin"
+                  ([ llvmPackages.llvm ]
+                   ++ optional stdenv.isDarwin llvmPackages.clang);
 in
-if paths == [] then ghc else
```

Comment:
This if-else clause prevents the wrapper from failing in case someone instantiates `ghcWithPackages` with an empty package, or with a package set that is non-empty but doesn't contain any actual Haskell libraries, i.e.:

  ghcWithPackages (hp: [hp.cabal-install])

Do these cases still work correctly with the LLVM patches applied?